### ISSUE TITLE
`ignore_blank_line` option does not affect "comment" textobject behavior

### ISSUE THREAD

**Issue by Valentin-Guillet** [echasnovski/mini.nvim#1379]
### Contributing guidelines

- [X] I have read [CONTRIBUTING.md](https://github.com/echasnovski/mini.nvim/blob/main/CONTRIBUTING.md)
- [X] I have read [CODE_OF_CONDUCT.md](https://github.com/echasnovski/mini.nvim/blob/main/CODE_OF_CONDUCT.md)
- [X] I have updated 'mini.nvim' to latest version

### Module(s)

mini.comment

### Description

The `mini.comment` plugin has an `ignore_blank_line` option to skip blank lines when commenting, but this setting doesn't modify the definition of the "comment" textobject.
For example, when two paragraphs are commented out (e.g. after `gc2ap`), the comment block definition only covers the first paragraph and not the whole commented section.

I feel like setting `ignore_blank_line` to true means that comment blocks also encompass blank lines, and IMHO the textobject should reflect this by skipping white lines. This behavior could either be the default or an additional plugin option.
For reference, this behavior is the default in tpope's vim-commentary.

To modify this behavior, I think a simple way would be to add a check for blank lines in the function returned by `H.make_comment_check` in the plugin:
`return function(line) return H.is_blank(line) or string.find(line, regex) ~= nil end`

### Neovim version

0.11.0

### Steps to reproduce

1. Set the `ignore_blank_line` option to true
2. Comment multiple paragraphs with `gc2ap`
3. Execute an action on a comment block `dgc`

### Expected behavior

Skip the blank links in the definition of comment textobjects when `ignore_blank_line = true`

### Actual behavior

A comment textobject stops at the next blank line.

**Comment by echasnovski** [comment]
Thanks for the suggestion!

At first glance, this might be a reasonable behavior. Current behavior is documented as [matter only when commenting](https://github.com/echasnovski/mini.nvim/blob/690a3b4c78c4956f7ecf770124b522d32084b872/doc/mini-comment.txt#L80) and not when uncommenting or computing textobject range.

An actual implementation might be a bit tricky because I'd say that only blanks "between" commented blocks should be part of the textobject. I.e. uncommented ranges before first commented line and after last commented line should be used.

I'll take a look, but I vaguely remember that textobject was not used affected by `config.ignore_blank_lines` for a reason.

**Comment by Valentin-Guillet** [comment]
Hey ! Sorry to bother, I had this problem again and it made me think about this issue.
Did you have time to look at this ?
I'd understand completely if you prefer not to modify the current behavior and close the issue, I just thought I'd ask whether I can expect changes on this.
Thanks a lot !

**Comment by echasnovski** [comment]
Yes, I am in the process of (gradual) backlog cleaning and this issue is on the list.

**Comment by Valentin-Guillet** [comment]
Great, no worries, thanks a lot !

**Comment by YouSame2** [comment]
can confirm just ran into this issue aswell today. caught me off gaurd not that big of a deal but heres where i ran into it, if it helps

```
return {
	"echasnovski/mini.nvim",
	version = "*",
	event = "VeryLazy",
	config = function()
		-- CUSTOM TEXTOBJECTS:
		-- local ai = require("mini.ai")
		-- local custom_txtobj = {}

		-- MINI.COMMENT
		local miniclue = require("mini.clue")
```

typing `dingc`, `digc`, `dgc` all did nothing when my cursor was above or inside the comment block.

mini.comments config:
-- MINI.COMMENT
require("mini.comment").setup()

**Comment by echasnovski** [comment]
> ... typing `dingc`, `digc`, `dgc` all did nothing when my cursor was above or inside the comment block.

Just FYI: `gc` is a self contained textobject. There is no `igc` or `ingc` as it is not a textobject from 'mini.ai'.

**Comment by echasnovski** [comment]
This now should be resolved on latest `main`. If `options.ignore_blank_line` is `true`, then blank lines in between commented lines are also considered part of the textobject.

**Comment by Valentin-Guillet** [comment]
I just tested it and it works perfectly, thanks a lot!
